src/14/HW/SortableTodoList.ts

```diff
@@ -0,0 +1,18 @@
+import { Note } from "./Note";
+import { TodoList } from "./TodoList";
+
+export class SortableTodoList extends TodoList {
+    public sortNotes(by: 'status' | 'createdAt'): Note[] {
+        const sortedNotes = [...this.getAllNotes()];
+
+        if (by === 'status') {
+            sortedNotes.sort((a, b) => a.status.localeCompare(b.status));
+        } else if (by === 'createdAt') {
```

Comment:
Handling this logic within a single method is ineffective, as it violates the Single Responsibility Principle. The logic here is too distinct, so it would be more beneficial to divide it into two separate methods.